sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs

```diff
@@ -166,40 +162,23 @@ internal static string GetUrlUsingHttpUrl(this AzMonList tagObjects)
             return url;
         }
 
-        internal static string GetUrlUsingHttpHost(this AzMonList tagObjects, string httpScheme, string defaultPort, string httpTarget)
+        internal static string GetUrlUsingHttpHost(this AzMonList tagObjects, string httpScheme, string httpTarget)
         {
-            string url = null;
             var httpHost = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpHost)?.ToString();
             if (!string.IsNullOrWhiteSpace(httpHost))
             {
-                string portSection = $":{defaultPort}";
-                if (httpHost.EndsWith(portSection, StringComparison.OrdinalIgnoreCase))
-                {
-                    var truncatedHost = httpHost.Substring(0, httpHost.IndexOf(portSection, StringComparison.OrdinalIgnoreCase));
-                    url = $"{httpScheme}://{truncatedHost}{httpTarget}";
-                }
-                else
-                {
-                    url = $"{httpScheme}://{httpHost}{httpTarget}";
-                }
+                var uribuilder = new UriBuilder(httpScheme, httpHost);
```

Comment:
I think we will need this one? as host can have port as per semantic convention example
<html>
<body>
<!--StartFragment-->

http.host | "example.com:8080"
-- | --


<!--EndFragment-->
</body>
</html>